zebra-consensus/src/transaction.rs

```diff
@@ -518,31 +499,66 @@ where
                 //.map_err(|e| BoxError::from(Box::new(e)))?;
             }
 
-            let _rsp = redjubjub_verifier
-                .ready_and()
-                .await?
-                .call((bvk, sapling_shielded_data.binding_sig, &shielded_sighash).into())
-                .boxed();
-
             // TODO: stop ignoring binding signature errors - #1939
-            // async_checks.push(rsp);
+            // async_checks.include(
+            //     primitives::redjubjub::VERIFIER
+            //         .clone()
+            //         .oneshot((bvk, sapling_shielded_data.binding_sig, &shielded_sighash).into()),
+            // );
         }
 
         Ok(async_checks)
     }
+}
+
+/// A set of asynchronous checks that should succeed.
+///
+/// A wrapper around [`FuturesUnordered`] with some auxiliary methods.
+struct AsyncChecks(FuturesUnordered<Pin<Box<dyn Future<Output = Result<(), BoxError>> + Send>>>);
+
+impl AsyncChecks {
+    /// Create an empty set of asynchronous checks.
+    pub fn new() -> Self {
+        AsyncChecks(FuturesUnordered::new())
+    }
+
+    /// Include a check in the set.
+    pub fn include(&mut self, check: impl Future<Output = Result<(), BoxError>> + Send + 'static) {
```

Comment:
Confusing API: `include` is confusingly similar to the `include!` and `include_str!` macros.

Using similar names to `FuturesUnordered` might be clearer:
https://docs.rs/futures/0.3.15/futures/stream/futures_unordered/struct.FuturesUnordered.html#method.push

Can you also say "push" or "extend" in the documentation of all the methods?

```suggestion
    pub fn push(&mut self, check: impl Future<Output = Result<(), BoxError>> + Send + 'static) {
```

Reply:
Done in [b3fd7af](https://github.com/jvff/zebra/commit/b3fd7af0314724b947f12b07ea9679e130b7cebb).